module/platform/Webots/src/Webots.cpp

```diff
@@ -222,34 +234,40 @@ namespace module::platform {
             else if (lvl == "FATAL") { this->log_level = NUClear::FATAL; }
             // clang-format on
 
+            clock_smoothing = config["clock_smoothing"].as<double>();
+
+            server_address = config["server_address"].as<std::string>();
+            server_port    = config["port"].as<std::string>();
+
             on<Watchdog<Webots, 5, std::chrono::seconds>>().then([this, config] {
                 // We haven't received any messages lately
-                log<NUClear::ERROR>("Connection timed out.");
-                setup_connection(config["server_address"].as<std::string>(), config["port"].as<std::string>());
+                log<NUClear::WARN>("Connection timed out. Attempting reconnect");
+                setup_connection();
             });
 
-            // Prime this reaction, so when only one ServoTargets is present, at least we have the other
-            emit(std::make_unique<RawSensors>());
-
             // Connect to the server
-            setup_connection(config["server_address"].as<std::string>(), config["port"].as<std::string>());
+            setup_connection();
         });
 
         // This trigger updates our current servo state
         on<Trigger<ServoTargets>, With<RawSensors>>().then([this](const ServoTargets& targets,
                                                                   const RawSensors& sensors) {
             // Loop through each of our commands
             for (const auto& target : targets.targets) {
+                // Get the difference between the current servo position and our servo target
                 const double diff = utility::math::angle::difference(
                     double(target.position),
                     utility::platform::getRawServo(target.id, sensors).present_position);
-
+                // Get the difference between the current time and the time the servo should reach its target
                 NUClear::clock::duration duration = target.time - NUClear::clock::now();
-                double speed                      = 0.0;
-                // If the clock has moved forward since the last update, then we can have a non-zero speed
-                if (duration.count() > 0) {
-                    speed = diff / std::chrono::duration_cast<std::chrono::duration<double>>(duration).count();
-                }
+
+                // If we have a positive duration, find the velocity.
+                // Otherwise, if the duration is negative or 0, the servo should have reached its position before now
+                // Because of this, we move the servo as fast as we can to reach the position.
+                // 5.236 == 50 rpm which is similar to the max speed of the servos
+                double speed = duration.count() > 0
+                                   ? diff / std::chrono::duration_cast<std::chrono::duration<double>>(duration).count()
```

Comment:
Pretty sure the `duration_cast` isn't needed here
```suggestion
                                   ? diff / std::chrono::duration<double>(duration).count()
```